File: lavcode-node/src/actions/password/dtos/get-password.dto.ts
import { V } from "@ipare/validator";
import { GetIconDto } from "../../../dtos/get-icon.dto";
import { IconEntity } from "../../../entities/icon.entity";
import { KeyValuePairEntity } from "../../../entities/key-value-pair.entity";
import { PasswordEntity } from "../../../entities/password-entity";
import { GetKeyValuePairDto } from "./get-key-value-pair.dto";

export class GetPasswordDto {
  @V().Required()
  id!: string;
  folderId!: string;
  @V().Required()
  title!: string;
  @V().Required()
  value!: string;
  @V().Required()
  remark!: string;
  @V().Required()
  order!: string;
  @V().Required()
  updatedAt!: number;

  @V().Required()
  icon?: GetIconDto;
  @V().Required().Items(GetKeyValuePairDto)
  keyValuePairs?: GetKeyValuePairDto[];

  public static fromEntity(
    entity: PasswordEntity,
    icon?: IconEntity,
    keyValuePairs?: KeyValuePairEntity[]
  ) {
    const result = new GetPasswordDto();
    result.id = entity._id;
    result.folderId = entity.folderId;
    result.title = entity.title;
    result.value = entity.value;
    result.remark = entity.remark;
    result.order = entity.order;
    result.updatedAt = entity.updatedAt;
    result.icon = icon ? GetIconDto.fromEntity(icon) : undefined;
    result.keyValuePairs = keyValuePairs?.map((item) =>
      GetKeyValuePairDto.fromEntity(item)
    );
    return result;
  }
}

File: src/Lavcode.Service.Api/Dtos/GetPasswordDto.cs
using Lavcode.Model;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Lavcode.Service.Api.Dtos
{
    public class GetPasswordDto
    {
        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("folderId")]
        public string FolderId { get; set; }

        [JsonProperty("title")]
        public string Title { get; set; }

        [JsonProperty("value")]
        public string Value { get; set; }

        [JsonProperty("remark")]
        public string Remark { get; set; }

        [JsonProperty("order")]
        public int Order { get; set; }

        [JsonProperty("updatedAt")]
        public long UpdatedAt { get; set; }

        [JsonProperty("icon")]
        public GetIconDto Icon { get; set; }

        [JsonProperty("keyValuePairs")]
        public IList<GetKeyValuePairDto> KeyValuePairs { get; set; }

        public PasswordModel ToModel()
        {
            return new PasswordModel()
            {
                Id = Id,
                FolderId = FolderId,
                Title = Title,
                Value = Value,
                Remark = Remark,
                Order = Order,
                UpdatedAt = DateTimeOffset.FromUnixTimeMilliseconds(UpdatedAt),
                Icon = Icon.ToModel(),
                KeyValuePairs = KeyValuePairs.Select(item => item.ToModel()).ToList()
            };
        }
    }
}
